### PR TITLE
Initial commit covering all existing functionality

### PR DESCRIPTION
**Pull request by thurstonsand**

### CODE REVIEW

File: hackerpet/__init__.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
+"""
```
Comment (BobSandberg) [inline hackerpet/__init__.py:1]: I typically like to see this kind of code broken out into separate module/files

File: hackerpet/__init__.py [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+                "weekday_from": Schedule._hm_format(self.weekday_from),
+                "weekday_to": Schedule._hm_format(self.weekday_to),
+                "weekend_from": Schedule._hm_format(self.weekend_from),
+                "weekend_to": Schedule._hm_format(self.weekend_to),
+            },
+            separators=(",", ":"),
+        )
+
+
+class Status:
```
Comment (BobSandberg) [inline hackerpet/__init__.py]: What would you think about breaking this out into a Status dataclass and a constructor that created it from the raw body? I understand that this is a recommended pattern by HomeAssistant, but not sure why it is a beneficial pattern.

File: hackerpet/__init__.py [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+                return "Matching More Colors"
+
+    @classmethod
+    def _missing_(cls: type, value: Any) -> "Game":
+        raise IllegalValue(Game, value)
+
+    @classmethod
+    def from_int_str(cls: type, game_str: str) -> "Game":
+        """convert str of int value to Game, or `GAME_UNKNOWN` if invalid"""
+        if game_str.isdigit():
```
Comment (BobSandberg) [inline hackerpet/__init__.py]: Would you consider raise or return an error as early as possible and then let normal flow and a single return value follow.

File: hackerpet/__init__.py (hunk 1 above)
Comment (thurstonsand) [reply hackerpet/__init__.py:1]: done

File: hackerpet/__init__.py (hunk 2 above)
Comment (thurstonsand) [reply hackerpet/__init__.py]: tried it. think it's better

File: hackerpet/__init__.py (hunk 3 above)
Comment (thurstonsand) [reply hackerpet/__init__.py]: sure, gave that a try. im always torn between this style and if/else style (tho secretly, i always prefer `Option.getOrElse` style when possible)

**Pull request merged**